afids_cnn/train.py

```diff
@@ -108,17 +110,32 @@ def gen_model() -> keras.Model:
 
 
 def fit_model(
-    model: keras.Model,
+    model: tf.keras.Model,
     new_train: Iterable[tuple[NDArray, NDArray]],
     model_out_path: os.PathLike[str] | str,
     loss_out_path: os.PathLike[str] | str | None,
+    epochs: int = 100,
+    steps_per_epoch: int = 50,
+    loss_fn: tf.losses.Loss | str = "mse",
+    optimizer: tf.optimizers.Optimizer | str | None = None,
+    metrics: list[tf.metrics.Metric | str] | None = None,
+    validation_data: Iterable[tuple[NDArray, NDArray]] | None = None,
 ):
+    if not optimizer:
+        optimizer = tf.keras.optimizers.Adam()
+    if metrics is None:
```

Comment:
Just for consistency, can this also be `if not metrics` or does it being a list change how this condition is interpreted

Reply:
That would prevent people from legitimately providing an empty list for the metrics. Not sure if that's valid in this framework though.

Reply:
Hmm, that is fair. My thought was that if you aren't providing a metric it'd be equivalent to `None` and should be falling back on whatever the default is in that situation. @ataha24 any insights on this?

Reply:
Here is the documentation reference: https://www.tensorflow.org/api_docs/python/tf/keras/Model#compile. Default is indeed None. Metrics are to be passed as a list, I think. 
<img width="886" alt="Screenshot 2023-09-12 at 11 20 06 AM" src="https://github.com/afids/afids-CNN/assets/46094728/49d016f3-8f78-4e32-b620-03175f99b31e">

Reply:
So by default, `metrics=None`, which would be the same as the `optimizer,` so should be able to handle the condition the same way. Alternatively, the input to `metrics` should always be a list with items in it (according to the typehint), so then this condition would not be met in those cases (and technically make the empty list an invalid input).

I'm not sure what the intended behaviour is suppose to be, because right now if `None` is passed in the workflow, it defaults to RMSE, but if an empty list is passed, I think the model compilation will actually cause tf to error out. If you change this to `if not metrics` then RMSE would be used in the case of both `None` or an empty list being passed in the workflow. _(Hopefully that made sense)_

Reply:
I'm open to whatever here -- I think the user passing an empty list is kind of an edge case where it's not really clear what the caller is trying to do, so I think just defaulting to RMSE is fine if you want that

Reply:
Yeah fair enough - @ataha24's call

Reply:
ok, I am fine with it defaulting to RMSE to make running workflow easier (i.e., less options to type in) as long as user can override that. Since narritive is landmark regression these metrics (i.e., MSE and RMSE) make sense rather than something like DICE.